colour-mapping/colour_mapping.py

```diff
@@ -0,0 +1,105 @@
+import dwavebinarycsp
+from hybrid.reference.kerberos import KerberosSampler
+
+from utilities import visualize_map
+
+
+class Province:
+    def __init__(self, name):
+        self.name = name
+        self.red = name + "_r"
+        self.green = name + "_g"
+        self.blue = name + "_b"
+        self.yellow = name + "_y"
+
+
+# Set up provinces
+bc = Province("bc")   # British Columbia
+ab = Province("ab")   # Alberta
+sk = Province("sk")   # Saskatchewan
+mb = Province("mb")   # Manitoba
+on = Province("on")   # Ontario
+qc = Province("qc")   # Quebec
+nl = Province("nl")   # Newfoundland and Labrador
+nb = Province("nb")   # New Brunswick
+pe = Province("pe")   # Prince Edward Island
+ns = Province("ns")   # Nova Scotia
+yt = Province("yt")   # Yukon
+nt = Province("nt")   # Northwest Territories
+nu = Province("nu")   # Nunavut
+
+provinces = [bc, ab, sk, mb, on, qc, nl, nb, pe, ns, yt, nt, nu]
+
+# Set up province neighbours (i.e. shares a border)
+neighbours = [(bc, ab),
+              (bc, nt),
+              (bc, yt),
+              (ab, sk),
+              (ab, nt),
+              (sk, mb),
+              (sk, nu),
+              (sk, nt),
+              (mb, on),
+              (mb, nu),
+              (on, qc),
+              (qc, nb),
+              (qc, nl),
+              (nb, ns),
+              (yt, nt),
+              (nt, nu)]
+
+# Initialize constraint satisfaction problem
+csp = dwavebinarycsp.ConstraintSatisfactionProblem(dwavebinarycsp.BINARY)
+not_both = {(0, 1), (1, 0), (0, 0)}
```

Comment:
could probably call this 'NAND'

Reply:
I find `not_both` to be easier to read later on (L65-68).